cognite/client/data_classes/capabilities.py

```diff
@@ -102,34 +109,126 @@ def dump(self, camel_case: bool = False) -> dict[str, Any]:
         capability_name = self._capability_name
         return {to_camel_case(capability_name) if camel_case else to_snake_case(capability_name): data}
 
+    def has_capability(self, other: Capability) -> bool:
+        if not isinstance(self, type(other)):
+            return False
+
+        if not other.scope.is_within(self.scope):
+            return False
+        missing_actions = set(other.actions) - set(self.actions)
+        return not missing_actions
+
+
+class ProjectScope(ABC):
+    name: ClassVar[str] = "projectScope"
+
+    @classmethod
+    def load(cls, resource: dict | str) -> ProjectsScope | AllProjectsScope:
+        resource = resource if isinstance(resource, dict) else load_yaml_or_json(resource)
+        ((name, data),) = resource.items()
+        if name != cls.name:
+            raise ValueError(f"Expected '{cls.name}', got '{name}'")
+
+        if "projects" in data:
+            return ProjectsScope(projects=data["projects"])
+        elif "allProjects" in data:
+            return AllProjectsScope()
+        raise ValueError(f"Unknown project scope: {data}")
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        if isinstance(self, AllProjectsScope):
+            return {self.name: {"allProjects": {}}}
+        elif isinstance(self, ProjectsScope):
+            return {self.name: {"projects": self.projects}}
+        raise ValueError(f"Unknown project scope: {self}")
+
+
+@dataclass(frozen=True)
+class AllProjectsScope(ProjectScope):
+    ...
+
+
+@dataclass(frozen=True)
+class ProjectsScope(ProjectScope):
+    projects: list[str]
+
+
+@dataclass
+class ProjectCapability(CogniteResource):
+    """This represents an capability scoped for a project(s)."""
+
+    capability: Capability
+
+    class Scope:
+        All = AllProjectsScope
+        Projects = ProjectScope
+
+    project_scope: AllProjectsScope | ProjectScope
+
+    @classmethod
+    def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> Self:
+        capability = next(key for key in resource if key != "projectScope")
+
+        return cls(
+            capability=Capability.load({capability: resource[capability]}),
+            project_scope=ProjectScope.load({ProjectScope.name: resource[ProjectScope.name]}),
+        )
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        dumped = self.capability.dump(camel_case=camel_case)
+        dumped.update(self.project_scope.dump(camel_case=camel_case))
+        return dumped
+
+
+class ProjectCapabilities(CogniteResourceList[ProjectCapability]):
+    _RESOURCE = ProjectCapability
+
+    def includes_capability(self, capability: Capability) -> bool:
```

Comment:
This will also ignore project scope for the capabilities. It needs to accept a project parameter.

Reply:
All in all I'm hesitant about adding this feature, as there are many intricacies wrt. how capabilities are handled, so having to duplicate that logic in SDKs is both error prone and a maintenance hassle. 

e.g. in data modeling you may implicitly have READ when the WRITE action is present, which is not reflected in the `has_capability` implementation for those capabilities.

Reply:
My point of adding it is to demonstrate why I want the `ProjectCapabilityList` to be there and not just a `list[ProjectCapability`. But this can be postponed. 

When it comes to whether we should do this feature at all, I know this is implemented in multiple projects which means it is worth having here. But I think we should have it well documented and clear that this implementation is not covering edge cases as the one you mentioned above. A basic implementation will cover 95% of all use cases.

Reply:
Moved this to a different PR, but kept the `has_capability` on `Capability`. It checks 1:1, so it is easy to keep as it is "always correct" as opposed to having capabilities spread over several groups.